### PR TITLE
Add Github package.json dependency version badge; affects [githubpackagejson githubmanifest npm]

### LINKED ISSUE #2259: dependency version badges for applications, from package.json
**Description**
I would love to be able to add `dependency` version badges from my `package.json` and have those badges change to let developers know that a dependency is outdated. I am aware of david-dm.org for example...

[peerDependencies Status](https://david-dm.org/guylepage3/lepage.cc)

But it would be amazing to have a badge for each of my dependencies..

### PR DESCRIPTION
**Pull request by paulmelnikow**
Close #2259 which is mostly about a `package.json` dependency badge.

~~Requires #2644.~~

### CODE REVIEW

File: services/github/github-package-json.service.js [hunk 1]
```diff
@@ -70,6 +78,99 @@ class GithubPackageJsonVersion extends ConditionalGithubAuthService {
...
+      base: 'github/package-json/dependency-version',
+      pattern:
+        ':user/:repo/:kind(dev|peer)?/:scope(@[^/]+)?/:packageName/:branch*',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'GitHub package.json dependency version',
```
Comment (calebcartwright) [inline services/github/github-package-json.service.js]: Thoughts on adding an indicator here for `prod` dependency? Or is it self explanatory/implicit enough?

File: services/github/github-package-json.tester.js [hunk 2]
```diff
@@ -46,3 +47,57 @@ t.create('Package array')
...
+  .get('/dependency-version/paulmelnikow/react-boxplot/simple-statistics.json')
+  .expectJSONTypes(
+    Joi.object({
+      name: 'simple-statistics',
+      value: semverRange,
+    })
+  )
+
+t.create('Scoped dependency')
+  .get('/dependency-version/zeit/next.js/dev/babel-eslint.json')
```
[the remark this answers is not in the document]
Comment (paulmelnikow) [reply services/github/github-package-json.tester.js]: Yes, scoped dependencies do, though `zeit/next.js` is the repo and `babel-eslint` is not scoped.

[the remark this answers is not in the document]
Comment (calebcartwright) [reply services/github/github-package-json.tester.js]: I'm a little confused then. I figured based on the test name that the target use case would be a scoped (dev) dependency of the target repo.

File: services/github/github-package-json.service.js [hunk 3]
```diff
@@ -70,6 +78,99 @@ class GithubPackageJsonVersion extends ConditionalGithubAuthService {
...
+    return {
+      label: 'dependency',
+    }
+  }
+
+  static render({ dependency, range }) {
+    return {
+      label: dependency,
+      message: range,
+      color: 'blue',
```
[the remark this answers is not in the document]
Comment (calebcartwright) [reply services/github/github-package-json.service.js]: Yeah that makes sense (on both counts).

File: services/github/github-package-json.tester.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (calebcartwright) [reply services/github/github-package-json.tester.js]: Could we change the dependency target to `@babel/preset-react` (https://github.com/zeit/next.js/blob/canary/package.json#L58) or maybe change the test name?

[the remark this answers is not in the document]
Comment (paulmelnikow) [reply services/github/github-package-json.tester.js]: Ah, right. I was trying to pick a dependency that wasn't likely to disappear and forgot to rename the test. Not sure if that is also true of `@babel/preset-react`…

File: services/package-json-helpers.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+}) {
+  let dependenciesOfKind
+  if (kind === 'peer') {
+    dependenciesOfKind = peerDependencies
+  } else if (kind === 'dev') {
+    dependenciesOfKind = devDependencies
+  } else if (kind === 'prod') {
+    dependenciesOfKind = dependencies
+  } else {
+    throw Error(`Not very kind: ${kind}`)
```
Comment (calebcartwright) [inline services/package-json-helpers.js]: 😆 The only time this would get hit is if a service accepts and/or passes along a bad dependency kind here right?

Comment (paulmelnikow) [reply services/package-json-helpers.js]: Thought I responded to this before. Yes! That would be a programmer error.

Comment (calebcartwright) [reply services/package-json-helpers.js]: Okay cool. Just wanted to make sure we weren't planning on surfacing this message up to end users 
👍

Comment (paulmelnikow) [reply services/package-json-helpers.js]: If it got thrown in production, they'd see **shields internal error**.

**Pull request merged**